src/auth.js

```diff
@@ -0,0 +1,223 @@
+    this.registrationAccessToken = opts.registrationAccessToken || opts.registration_access_token || ''
+
+    if (opts.redirect_uris) {
+      this.redirectURI = opts.redirect_uris[0] || ''
+    } else {
+      this.redirectURI = opts.redirectURI || ''
+    }
+
+    this.softwareID = opts.softwareID || opts.software_id || ''
+    this.softwareVersion = opts.softwareVersion || opts.software_version || ''
+    this.clientName = opts.clientName || opts.client_name || ''
+    this.clientKind = opts.clientKind || opts.client_kind || ''
+    this.clientURI = opts.clientURI || opts.client_uri || ''
+
+    this.logoURI = opts.logoURI || opts.logo_uri || ''
+    this.policyURI = opts.policyURI || opts.policy_uri || ''
+
+    if (this.url === '') {
+      throw new Error('Missing url attribute')
+    }
+    if (this.redirectURI === '') {
+      throw new Error('Missing redirectURI field')
+    }
+    if (this.softwareID === '') {
+      throw new Error('Missing softwareID field')
+    }
+    if (this.clientName === '') {
+      throw new Error('Missing clientName field')
+    }
+
+    while (this.url[this.url.length - 1] === '/') {
+      this.url = this.url.slice(0, -1)
+    }
+  }
+
+  isRegistered () {
+    return this.clientID !== ''
+  }
+
+  toRegisterJSON () {
+    return {
+      redirect_uris: [this.redirectURI],
+      software_id: this.softwareID,
+      software_version: this.softwareVersion,
+      client_name: this.clientName,
+      client_kind: this.clientKind,
+      client_uri: this.clientURI,
+      logo_uri: this.logoURI,
+      policy_uri: this.policyURI
+    }
+  }
+}
+
+export class Token {
+  constructor (opts) {
+    this.tokenType = opts.tokenType || opts.token_type
+    this.accessToken = opts.accessToken || opts.access_token
+    this.refreshToken = opts.refreshToken || opts.refresh_token
+    this.scope = opts.scope
+  }
+
+  toAuthHeader () {
+    return 'Bearer ' + this.accessToken
+  }
+}
+
+export function registerClient (client) {
+  if (!(client instanceof Client)) {
+    client = new Client(client.url, client)
+  }
+  if (client.isRegistered()) {
+    return Promise.reject(new Error('Client already registered'))
+  }
+  return fetch(`${client.url}/auth/register`, {
+    method: 'POST',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: JSON.stringify(client.toRegisterJSON())
+  })
+    .then(handleResponse)
+    .then((data) => new Client(client.url, data))
+}
+
+// getClient will retrive the registered client informations from the server.
+export function getClient (client, token) {
+  if (!(client instanceof Client)) {
+    client = new Client(client.url, client)
+  }
+  if (!client.isRegistered()) {
+    return Promise.reject(new Error('Client not registered'))
+  }
+  return fetch(`${client.url}/auth/register/${client.clientID}`, {
+    method: 'GET',
+    headers: {
+      'Accept': 'application/json',
+      'Authorization': token.toAuthHeader()
```

Comment:
Nope, it's not the access_token that is used here, but the registration_access_token!